server/BootstrapServer.java

```diff
@@ -177,23 +203,39 @@ public static void main(String args[])
                     String ip = st.nextToken();
                     int port = Integer.parseInt(st.nextToken());
                     String username = st.nextToken();
+                    boolean found = false;
                     for (int i=0; i<nodes.size(); i++) {
                         if (nodes.get(i).getPort() == port) {
                             nodes.remove(i);
-                            String reply = "0012 UNROK 0";
+                            reply = "0012 UNROK 0";
                             DatagramPacket dpReply = new DatagramPacket(reply.getBytes() , reply.getBytes().length , incoming.getAddress() , incoming.getPort());
                             sock.send(dpReply);
+                            found = true;
+                            break;
```

Comment:
hope this is intentional?
why does this show as a change?

Reply:
@PasinduUkwatta do we need this flag here? Breaking the loop is it a good idea?

Reply:
I added the flag to check whether the selected node is available or not, the loop break added due to stop the for loop after find the node. Else the loop will continue until whole node  size.